pkg/destroy/ibmcloud/cloudobjectstorage.go

```diff
@@ -0,0 +1,117 @@
+package ibmcloud
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+const cosTypeName = "cos instance"
+
+// listCOSInstances lists COS service instances
+func (o *ClusterUninstaller) listCOSInstances() (cloudResources, error) {
+	o.Logger.Debugf("Listing COS instances")
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+
+	resourceGroupID, err := o.ResourceGroupID()
+	if err != nil {
+		return nil, err
+	}
+
+	options := o.controllerSvc.NewListResourceInstancesOptions()
+	options.SetResourceGroupID(resourceGroupID)
+	resources, _, err := o.controllerSvc.ListResourceInstancesWithContext(ctx, options)
```

Comment:
We may also want to set a `Limit` option here (Number of items returned)
...not sure what the default is, but might be a good idea.


Also, if there are more than `100` Instances in the ResourceGroup there is a possibility we may not find the `cosInstanceName`. Might be a good idea to use `next_url` from Response and request the next page of Instances.

Reply:
OR you can use `ResourceID` **"is.cos"** or `Type` option to get a filtered response. Will save time while iterating over the list.

Reply:
"is.cos"? Wouldn't this have to be the COS service GUID?

Reply:
Ideally, we'd handle the pagination, but the vast majority of the time, there would never be >100 COS instances within the cluster resource group. Something very weird would have happened in that case. So, I'll make sure we only pull COS instances and not all types of RC Resources.

Reply:
I meant `>100` all instances under a ResourceGroup.

Yup, only pulling COS would be perfect.

You can pass in `ResourceID`: e.g.`is.volume`, `is.image`,` is.vpc` - Not sure, but there might be `is.cos`